lms/js_config_types.py

```diff
@@ -4,7 +4,28 @@
 Making this a top level module to avoid circular dependency problems.
 """
 
-from typing import NotRequired, TypedDict
+from typing import Literal, NotRequired, TypedDict
+
+
+class AutoGradingConfig(TypedDict):
+    grading_type: Literal["all_or_nothing", "scaled"] | None
+    """
+    - all_or_nothing: students need to meet a minimum value, making them get
+                      either 0% or 100%
+    - scaled: students may get a proportional grade based on the amount of
+              annotations. If requirement is 4, and they created 3, they'll
+              get a 75%
+    """
+
+    activity_calculation: Literal["cumulative", "separately"] | None
+    """
+    - cumulative: both annotations and replies will be counted together for
+                  the grade calculation
+    - separately: students will have different annotation and reply goals.
+    """
+
+    required_annotations: int | None
```

Comment:
The FE will always send this, either to represent only required annotations when `activity_calculation` is `separate`, or by representing required annotations and replies when it is `cumulative`.

Reply:
Changed it accordingly :+1: